app/code/Magento/PageCache/Observer/FlushCacheByTags.php

```diff
@@ -40,11 +40,19 @@ class FlushCacheByTags implements ObserverInterface
     /**
      * @param \Magento\PageCache\Model\Config $config
      * @param \Magento\Framework\App\PageCache\Cache $cache
+     * @param \Magento\Framework\App\Cache\Tag\Resolver $tagResolver
+     * @param \Magento\PageCache\Model\Cache\Type $fullPageCache
      */
-    public function __construct(\Magento\PageCache\Model\Config $config, \Magento\Framework\App\PageCache\Cache $cache)
-    {
+    public function __construct(
+        \Magento\PageCache\Model\Config $config,
+        \Magento\Framework\App\PageCache\Cache $cache,
+        \Magento\Framework\App\Cache\Tag\Resolver $tagResolver,
```

Comment:
For backward to maintain backward compatibility, you might want to change the constructor parameters a little bit.

```php
public function __construct(
    ...
    \Magento\Framework\App\Cache\Tag\Resolver $tagResolver = null,
    \Magento\PageCache\Model\Cache\Type $fullPageCache = null
) {
    ....
    $this->tagResolver   = $tagResolver ?: ObjectManager::getInstance()->get(\Magento\Framework\App\Cache\Tag\Resolver::class);
    $this->fullPageCache = $fullPageCache ?: ObjectManager::getInstance()->get(\Magento\PageCache\Model\Cache\Type::class);
}
```

Devdocs:
https://devdocs.magento.com/guides/v2.3/extension-dev-guide/object-manager.html#exceptions

Example:
\Magento\Contact\Model\Mail::__construct

Reply:
Hi @atishgoswami, I don't think we need to follow backward compatibility for the scope of `2.4-develop`: https://github.com/magento/magento2/pull/25875#discussion_r374904047

Reply:
O my bad .... seems like standards have changed again ..... thanks for pointing that out ..... 
need to update certain PR's myself now 😞